main/src/ca/uwaterloo/flix/MutationTester.scala

```diff
@@ -0,0 +1,278 @@
+package ca.uwaterloo.flix
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.Ast.Constant
+import ca.uwaterloo.flix.language.ast.SemanticOp.{BoolOp, CharOp, Float32Op, Float64Op, Int16Op, Int32Op, Int64Op, Int8Op, StringOp}
+import ca.uwaterloo.flix.language.ast.{Ast, SemanticOp, Symbol, TypedAst}
+import ca.uwaterloo.flix.language.ast.TypedAst.Expr
+import ca.uwaterloo.flix.tools.Tester
+import ca.uwaterloo.flix.util._
+import org.json4s.reflect.fail
+
+import java.io.File
+
+
+object MutationTester {
+    def run(files: Seq[File], options: Options, tester: String, testee: String): Unit = {
+        val flix = new Flix
+        flix.setOptions(options)
+        files.map(file => flix.addFlix(file.toPath))
+        flix.addSourceCode("testee",
+            s"""
+            |def main(): Unit \\ IO =
+            |    println(Test.one())
+            |def two(): Int32 = 2
+            |mod Fake {
+            |    pub def fake(): Int32 = 99
+            |}
+            |mod Test {
+            |    pub def one(): Int32 = 1
+            |    pub def ite(b: Bool): Int32 = if(b) 1 else 2
+            |}""".stripMargin)
+
+        flix.addSourceCode("tester",
+            s"""
+            |mod MainTest {
+            |    @test
+            |    def testConstant(): Bool =
+            |        Assert.eq(1, Test.one())
+            |    @test
+            |    def testIte(): Bool =
+            |       Assert.eq(1, Test.ite(true))
+            |}""".stripMargin)
+        val root = flix.check().unsafeGet
+        val start = System.nanoTime()
+        val root1 = mutate(root, testee)
+        val end = System.nanoTime() - start
+        println("time to generate mutations: " + end.toString)
+        runMutations(flix, root, root1)
+        /**
+        val result = root1.map(r => flix.codeGen(r).unsafeGet)
+        val tests = result.map(res => res.getTests)
+        */
+    }
+
+    def runMutations(flix: Flix, root: TypedAst.Root, mutatedDefs: List[(Symbol.DefnSym, List[TypedAst.Def])]): Unit = {
```

Comment:
This function looks like a reasonable start, but the main loop can probably be broken down into some smaller functions.

Reply:
E.g.:


- runAllTests
- runTest